src/bridge/cockpitpipechannel.c

```diff
@@ -105,6 +107,27 @@ process_pipe_buffer (CockpitPipeChannel *self,
 }
 
 static gboolean
+cockpit_pipe_channel_read_window_size_options (JsonObject *options,
+                                               gushort default_rows,
+                                               gushort default_cols,
+                                               gushort *rowsp,
+                                               gushort *colsp)
+{
+  gint64 rows, cols;
+
+  if (cockpit_json_get_int (options, "window_rows", default_rows, &rows) &&
+      cockpit_json_get_int (options, "window_cols", default_cols, &cols) &&
+      rows >= 0 && rows < G_MAXUINT16 && cols >= 0 && cols < G_MAXUINT16)
+    {
```

Comment:
When setting an unsupported value here, we should do a g_message() about it. In addition these limitations shuold be documented in doc/protocol.md when you document the window.rows and window.cols options.